packages/flutter/test/widgets/dismissible_test.dart

```diff
@@ -558,4 +614,29 @@ void main() {
     final RenderBox backgroundBox = tester.firstRenderObject(find.text('background'));
     expect(backgroundBox.size.height, equals(100.0));
   });
-}
+
+  testWidgets('Horizontal nudge less than threshold', (WidgetTester tester) async {
+    await tester.pumpWidget(buildTest());
+    expect(dismissedItems, isEmpty);
+
+    await rollbackItem(tester, 0, gestureDirection: AxisDirection.left);
+    expect(find.text('0'), findsOneWidget);
+    expect(dismissedItems, isEmpty);
+
+    await rollbackItem(tester, 1, gestureDirection: AxisDirection.right);
+    expect(find.text('1'), findsOneWidget);
+    expect(dismissedItems, isEmpty);
+  });
+  testWidgets('Vertical nudge less than threshold', (WidgetTester tester) async {
+    await tester.pumpWidget(buildTest());
+    expect(dismissedItems, isEmpty);
+
+    await rollbackItem(tester, 0, gestureDirection: AxisDirection.down);
+    expect(find.text('0'), findsOneWidget);
+    expect(dismissedItems, isEmpty);
+
+    await rollbackItem(tester, 1, gestureDirection: AxisDirection.up);
+    expect(find.text('1'), findsOneWidget);
+    expect(dismissedItems, isEmpty);
+  });
+}
```

Comment:
maybe i misunderstand the test but it doesn't seem to me that it verifies that things actually go diagonally or that they actually take rollbackDuration time to go away. For example, if someone replaced `widget.crossAxisEndOffset` with `0.0` in `_updateMoveAnimation`, would the test fail? Similarly, would your test fail if the duration was not precisely honoured, but instead e.g. the rollback happened in half the specified time? or double the specified time?

Reply:
You are right again, what all it does is check if the widget is still there after giving it a small nudge. For `rollbackDuration`, I am trying to find a more specific way to test the features but I am caught up in it. What I have in mind now is to check the duration once the widget is released from drag, but I can't seem to find a way to do it. And for `crossAxisEndOffset`, I am trying to get the `topLeft` or `topRight` offset and check, but I am not sure if it is the right way to go ahead. I haven't had success doing it yet.

Reply:
For the duration, you can just pump a duration less than the rollback, verify it hasn't finished rolling back, then pump a duration that takes you past the end of the rollback, and verify that it _has_ finished.

For the positions, yet, you can use tester.getTopLeft and so forth to find coordinates.